src/App/App.xaml.cs

```diff
@@ -335,6 +337,14 @@ public async Task UpdateThemeAsync()
             });
         }
 
+        private async Task ClearSensitiveFieldsAsync()
+        {
+            await Device.InvokeOnMainThreadAsync(() =>
+            {
+                _messagingService.Send(Constants.ClearSensitiveFields);
+            });
+        }
```

Comment:
This shouldn't need to be invoked on the main thread, the receiver of the message should be the one who does that, or isn't the message being received because of being in another thread?

Reply:
The latter I believe.  I'm unable to update the UI in Android without invoking this on the main thread.

Reply:
Have you tried debugging it to see if the message arrives or where does it stop? Because from what I can see the receiver is already invoking on the main thread.
I'd be very helpful to see where it stops so we can provide a better way to handle that instead of the caller invoking on the main thread just to send a message.

Reply:
Yessir, the message does not arrive in `LoginPage` or `LockPage` unless invoked on the main thread from `App`.  I'm able to follow the transmission into `BroadcasterService` and through the for loop in `Send`.  The only time `LockPage` is included as a `sub` in the resume flow is when the message is sent from the main thread (so `updatedTheme` goes to `LockPage` but `clearSensitiveFields` doesn't).  I originally assumed it was a UI related limitation (hence sending `updatedTheme` from the main thread as well) but now I'm not really sure.  Thoughts?

Reply:
I think the problem is that the message is sent before the registration is done. And when you do it on the main thread it'll wait until the main thread finishes with the current work to send the message and that's why is working in that case.
I'm not quite sure if it'll work on all devices but even though I don't like much this solution the other one is to change the messenger/broadcast to have a special send method that queues the message if there are no subscribers at the moment for a specific id or something like that. So when the subscriber with that Id subscribes, the queue is checked and the message is sent accordingly in time.
what do you think?

Reply:
Sounds useful, though it probably deserves it's own PR since it would require refactoring other messages sent in the same manner.

Reply:
Agree 😄 approving this then